yt/data_objects/static_output.py

```diff
@@ -125,23 +125,21 @@ def __init__(self, display_array=False):
             self.display_array = False
 
     def __get__(self, instance, owner):
-        if not instance:
-            return None
         ret = self.data.get(instance, None)
-        try:
-            ret = ret.copy()
-        except AttributeError:
-            pass
-        if self.display_array:
-            try:
-                ret._ipython_display_ = functools.partial(_wrap_display_ytarray, ret)
-            # This will error out if the items have yet to be turned into
-            # YTArrays, in which case we just let it go.
-            except AttributeError:
-                pass
         return ret
 
     def __set__(self, instance, value):
+        if self.display_array and find_spec("ipywidgets") is not None:
+            try:
+                value._ipython_display_ = functools.partial(
+                    _wrap_display_ytarray, value
+                )
+            except AttributeError:
```

Comment:
I don't think the try/except serves any purpose after your refactor. Am I missing anything ?

Reply:
Yeah, I think I got confused doing the merge

Reply:
Seeing the failures, I understand how wrong I was to assume that setting a new attribute on any object is always valid; ndarrays and other objects might in fact use protective mechanisms like `__slots__` to prevent it. Anyway, sorry about the confusion!

Reply:
I'd suggest adding a comment to explain this, because my memory apparently cannot retain this reasoning for a whole week 😅